src/Criteo.OpenApi.Comparator/Comparators/OperationComparator.cs

```diff
@@ -257,30 +257,75 @@ private static int FindParameterIndex(OpenApiParameter parameter, IList<OpenApiP
         /// <returns>Swagger Parameter if found; otherwise null</returns>
         private static OpenApiParameter FindParameter(
-            string name,
+            OpenApiParameter key,
             IEnumerable<OpenApiParameter> operationParameters,
             IDictionary<string, OpenApiParameter> documentParameters)
         {
+            string name = key.Name;
             if (name == null || operationParameters == null)
                 return null;
 
+            var candidateParameters = new List<OpenApiParameter>();
             foreach (var parameter in operationParameters)
             {
                 if (name.Equals(parameter.Name))
-                    return parameter;
+                    candidateParameters.Add(parameter);
+                else
+                {
+                    var referencedParameter = parameter.Reference.Resolve(documentParameters);
 
-                var referencedParameter = parameter.Reference.Resolve(documentParameters);
+                    if (referencedParameter != null && name.Equals(referencedParameter.Name))
+                        candidateParameters.Add(referencedParameter);
+                }
+            }
 
-                if (referencedParameter != null && name.Equals(referencedParameter.Name))
-                    return referencedParameter;
+            if (!candidateParameters.Any())
+                return null;
+
+            if(candidateParameters.Count == 0)
+                return candidateParameters[0];
+
+            return GetClosestFunctionally(key, candidateParameters);
+        }
+
+        /// <summary>
+        /// Determines the most similar parameter to the key parameter based on functional similarity.
+        /// </summary>
+        /// <param name="key">The parameter to score against for functional similarity</param>
+        /// <param name="candidateParameters">The candidates. One of these will be selected as the most similar</param>
+        /// <returns>The most similar parameter functionally to the key parameter</returns>
+        private static OpenApiParameter GetClosestFunctionally(OpenApiParameter key, List<OpenApiParameter> candidateParameters)
+        {
+            OpenApiParameter bestMatch = null;
+            var bestMatchScore = 0;
+
+            foreach (var candidate in candidateParameters)
+            {
+                var score = 0;
+
+                score += key.Reference == candidate.Reference ? 1 : 0;
+                score += key.In == candidate.In ? 1 : 0;
+                score += key.Required == candidate.Required ? 1 : 0;
+                score += key.Deprecated == candidate.Deprecated ? 1 : 0;
+                score += key.AllowEmptyValue == candidate.AllowEmptyValue ? 1 : 0;
+                score += key.Style == candidate.Style ? 1 : 0;
+                score += key.Explode == candidate.Explode ? 1 : 0;
+                score += key.AllowReserved == candidate.AllowReserved ? 1 : 0;
```

Comment:
That's an interesting take, but I worry that it would make the comparison less predictable.
Could you add some tests to showcase a bit what the behavior could be?

Also, I feel this would miss the most important similarity for path parameters: the order in which they are present in the path.

Reply:
I was thinking this also. I'm wondering if a better approach might be the index of the parameter in the path, but I think the order of header parameters and query parameters doesn't actually matter. Also if an optional parameter is added, it will move a parameter. So I thought that parameter index wouldn't be as useful in the end as values. 

This comparison is a best guess on my part, so I'm open to better comparison methods. My actual intention is to find the parameter that EXACTLY matches the old parameter, no matter where it appears in the parameters list. Then, if that does not exist, take a reasonable remaining candidate. This will maintain the behaviour that catch legitimate `1015`s, without falsely identifying some. 

Anyway, I'll write a few more tests to make sure that there is a decent rail to prove it works properly.